components/controls/Tooltip.tsx

```diff
@@ -252,8 +286,85 @@ export class TooltipInner extends React.Component<TooltipProps, State> {
     this.handleMouseLeave();
   };
 
-  render() {
+  needsFlipping = (leftFix: number, topFix: number) => {
+    // We can live with a tooltip that's slightly positioned over the toggle
+    // node. Only trigger if it really starts overlapping, as the re-positioning
+    // is quite expensive, needing 2 re-renders.
+    const threshold = this.context.theme.rawSizes.grid;
```

Comment:
I think this threshold might be a bit too big.

![tooltip_threshol](https://user-images.githubusercontent.com/51169196/63592654-5f959580-c5b2-11e9-91f0-b2ac023ed436.gif)

Reply:
Ha! That's a `HelpToolTip`, they lazy load their content 😑 . So the height is not correctly computed, and the positioning is off.